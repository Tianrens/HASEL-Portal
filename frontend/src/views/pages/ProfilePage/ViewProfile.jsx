import { React, useState } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import styles from './Profile.module.scss';
import StyledHeader from '../../components/text/StyledHeader';
import TopBarPageTemplate from '../../components/templates/TopBarPageTemplate/TopBarPageTemplate';
import UserDetails from './UserDetails';
import BottomButtons from '../../components/buttons/BottomButtons';
import { discardResourceMessage, editResourceMessage } from '../../../config/ModalMessages';
import { patchUtil } from '../../../util/apiUtil';
import { useGet } from '../../../hooks/useGet';
import LoadingWheelDiv from '../../components/LoadingWheel/LoadingWheelDiv';

const ViewProfile = () => {
    const { userId } = useParams();
    const history = useHistory();

    const user = useGet(`/api/user/${userId}`).data;
    const [type, setType] = useState(user?.type);
    const typeChanged = type !== user?.type;
    const invalidChange = type === 'SUPERADMIN' && user?.type === 'ADMIN';

    return (
        <TopBarPageTemplate>
            {!user ? (
                <LoadingWheelDiv />
            ) : (
                <div className={styles.fadeInAnimation}>
                    <StyledHeader left>{`${user?.firstName}'s Account`}</StyledHeader>
                    <UserDetails user={user} adminView updateType={(newType) => setType(newType)} />
                    <BottomButtons
                        onAccept={patchUtil('user', { type }, userId, () => history.goBack())}
                        onDeny={() => history.goBack()}
                        acceptText='Confirm Changes'
                        denyText='Cancel'
                        denyMessage={discardResourceMessage('user')}
                        acceptMessage={editResourceMessage('user')}
                        acceptDisabled={!typeChanged || invalidChange}
                    />
                </div>
            )}
        </TopBarPageTemplate>
    );
};

export default ViewProfile;
