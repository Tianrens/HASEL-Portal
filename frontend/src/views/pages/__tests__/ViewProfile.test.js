import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import { MemoryRouter } from 'react-router-dom';
import { SnackbarProvider } from 'notistack';
import ViewProfile from '../ProfilePage/ViewProfile';

test('Edit Booking page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <SnackbarProvider>
                <MemoryRouter>
                    <ViewProfile />
                </MemoryRouter>
            </SnackbarProvider>
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
