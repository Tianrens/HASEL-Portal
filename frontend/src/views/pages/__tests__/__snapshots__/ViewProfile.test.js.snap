// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Booking page renders properly 1`] = `
Array [
  <div
    className="fillTopbarSpace"
  >
    <div
      className="navbar"
    >
      <div
        className="patternBgLayer"
      >
        <div
          className="bgLayer"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="University of Auckland"
              className="uoaLogo"
              src="uoaLogo.svg"
            />
            <div
              className="spacer"
            />
            <div
              className="title"
            >
              HASEL Portal
            </div>
          </a>
          <div
            className="pushSpacer"
          />
          <div
            className="horizontalNavLinks"
          >
            <div
              className="linksWrapper"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                <button
                  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root wrapper css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <img
                    alt="My Account"
                    className="navIcon"
                    src="id.svg"
                  />
                  <div
                    className="spacer"
                  />
                  <div
                    className="title"
                  >
                    My Account
                  </div>
                </button>
              </a>
            </div>
          </div>
          <div
            className="drawerWrapper"
            onClick={[Function]}
            onTouchEnd={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge drawerButton css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DehazeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 15.5v2h20v-2H2zm0-5v2h20v-2H2zm0-5v2h20v-2H2z"
                />
              </svg>
            </button>
            <div
              className="MuiDrawer-root MuiDrawer-docked drawer css-7ik032-MuiDrawer-docked"
            >
              <div
                className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
                style={
                  Object {
                    "visibility": "hidden",
                  }
                }
              >
                <div
                  className="drawerPatternBg"
                >
                  <div
                    className="drawer"
                  >
                    <div>
                      <button
                        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge drawerButton css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CloseIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      className="linksWrapper"
                    >
                      <a
                        href="/user"
                        onClick={[Function]}
                      >
                        <button
                          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root wrapper css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                          disabled={false}
                          onBlur={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          type="button"
                        >
                          <img
                            alt="My Account"
                            className="navIcon"
                            src="id.svg"
                          />
                          <div
                            className="spacer"
                          />
                          <div
                            className="title"
                          >
                            My Account
                          </div>
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="content"
  >
    <div
      className="container"
    >
      <div
        className="container"
      >
        <div
          className="chase"
        >
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
          <div
            className="chaseDot"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
